File: php/features.php
<?php
declare(strict_types=1);

// FunisgoStreaming SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FunisgoStreamingFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FunisgoStreamingBaseFeature();
            case "test":
                return new FunisgoStreamingTestFeature();
            default:
                return new FunisgoStreamingBaseFeature();
        }
    }
}

File: py/features.py
# FunisgoStreaming SDK feature factory

from feature.base_feature import FunisgoStreamingBaseFeature
from feature.test_feature import FunisgoStreamingTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FunisgoStreamingBaseFeature(),
        "test": lambda: FunisgoStreamingTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
